lib/salestation/rspec.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'rspec/expectations'
+require 'salestation/app'
+
+require_relative './rspec/failure_matcher'
+require_relative './rspec/glia_input_validation_error_matcher'
+
+module Salestation
+  module RSpec
+    # Usage:
+    #
+    # In your RSpec configuration first include the matchers:
+    #
+    #   RSpec.configure do |config|
+    #     config.include Salestation::RSpec::Matchers
+    #   end
+    #
+    # Then you can use the matchers like this:
+    #
+    #   expect(result).to be_failure
+    #     .with_conflict
+    #     .containing(hash_including(message: 'Oh noes'))
+    #
+    # or when using Glia Errors:
+    #
+    #   expect(result).to be_failure
+    #     .with_requested_resource_not_found
+    #     .based_on(Glia::Errors::ResourceNotFoundError.new(resource: :user))
+    #
+    # or when matching input errors from Glia Errors:
+    #
+    #   expect(result).to be_failure
+    #     .with_invalid_input
+    #     .based_on(glia_input_validation_error.on(:name).with_type(Glia::Errors::INVALID_VALUE_ERROR))
```

Comment:
From what I see this only allows validating one field, or not?

Would this be possible?
```
glia_input_validation_error
 .on(:name).with_type(Glia::Errors::INVALID_VALUE_ERROR)
 .on(:name).with_type(Glia::Errors::INVALID_VALUE_ERROR)
```

Reply:
Currently it is possible to validate only one field. You could use multiple error types though:
```
glia_input_validation_error
 .on(:name).with_type(Glia::Errors::INVALID_VALUE_ERROR, Glia::Errors::SOMETHING_ELSE)
```

Your proposal (I assume the last line would have a different field name) would be possible to implement but I didn't see a need for that immediately.

Reply:
I think it could be quite often that multiple fields are tested like https://github.com/salemove/phone-number-manager/blob/master/spec/phone_number_manager/create_incoming_phone_number_spec.rb#L77
https://github.com/salemove/engagement-registry/blob/9a310c667c88dd86b31315dfc44cce8786762c10/spec/app/engagement/handle_visitor_call_end_spec.rb#L70

Reply:
In general, I would recommend testing fields one by one for better diffs and more robust tests, but I do see that it can be helpful sometimes. Added it here: https://github.com/salemove/salestation/compare/e51b07d36ba9caec4e985c085eedc7a11bc506a4..a1843e7f2c11f1d51cdb053beed8fd7ede202d49